blog: lw7 o11y day (#13451)
supabase/supabase: apps/www/_blog/2023-04-10-supabase-logs-self-hosted.mdx

---

> title: Supabase Logs are now self-hosted

Reviewer: Shouldn't the title be something like:

```suggestion
title: Supabase Logs can now be self-hosted
```

Just reading this it gave me the impression that there won't have a cloud offering anymore.

launchweek: 7
tags:
  - launch-week
  - o11y
date: '2023-04-10'
to_depth: 3
author: chasers,ziinc
---

Today, we’re releasing Supabase Logs for both self-hosted users and CLI development.

## Logflare Introduction

Since [Logflare joined Supabase](https://supabase.com/blog/supabase-acquires-logflare) over a year ago it’s been quietly handling over 1 billion log events every day. These events come from various tools in the Supabase infrastructure - the API gateway, Postgres databases, Storage, Edge Functions, Auth, and Realtime.

Logflare is a multi-node, highly available Elixir cluster, ingesting the log events and storing them into BigQuery for Supabase and Logflare’s customers. On average, the cluster has 6 nodes handling every spike our customers throw at it.

> To expose log data to customers, we leverage Logflare Endpoints. This provides an HTTP integration into Supabase Studio, powering the log query UIs and most timeseries charts. These charts live across the studio, such as the project home page and the new API reports.

Reviewer: nit:

```suggestion
To expose log data to customers, we leverage Logflare Endpoints. This provides an HTTP integration into Supabase Studio, powering the log query UIs and most time-series charts. These charts live across the studio, such as the project home page and the new API reports.
```

{/* <!--  add chart screenshot --> */}

## Self-hosting Logflare

Logflare was available under a BSL license prior to joining Supabase. We’ve since changed the license to [Apache 2.0](https://github.com/Logflare/logflare/blob/staging/LICENSE.md), aligning it with our open source philosophy.

> In the past few months we’ve made Logflare more developer-friendly for local development and self-hosting. While you’re building a project, you can view and query your logs from any Supabase service, just as you would in our cloud Platform.

Reviewer: Did you mean to capitalize?

```suggestion
In the past few months we’ve made Logflare more developer-friendly for local development and self-hosting. While you’re building a project, you can view and query your logs from any Supabase service, just as you would in our cloud platform.
```

![Realtime local logs](/images/blog/lw7-o11y/realtime-logs-screenshot.png)

<div className="bg-gray-300 rounded-lg px-6 py-2 italic">

📢 Check out the [new self-hosting docs](https://supabase.com/docs/reference/self-hosting-analytics/introduction) to get Logflare up and running as your analytics server.

</div>

It currently supports a BigQuery backend, and we are actively working on supporting more.

## The Ingestion Pipeline

> Logflare receives Supabase log events via multiple methods. Services like Postgres use [Vector](vector.dev/) to clean and forward log events to our the Logflare ingest API. Other services such as Realtime and Storage utilize [native Logflare integrations](https://github.com/Logflare/logflare#integrations) to send the log events directly. These then get processed and streamed into BigQuery.

Reviewer: ```suggestion
Logflare receives Supabase log events via multiple methods. Services like Postgres use [Vector](vector.dev/) to clean and forward log events to the Logflare ingest API. Other services such as Realtime and Storage utilize [native Logflare integrations](https://github.com/Logflare/logflare#integrations) to send the log events directly. These then get processed and streamed into BigQuery.
```
